File: sv/iamp-lib-usm/src/main/java/com/infosys/icets/iamp/usm/domain/UsmModule.java
/**
 * @ 2023 Infosys Limited, Bangalore, India. All Rights Reserved.
 * Version: 1.0
 * Except for any free or open source software components embedded in this Infosys proprietary software program (Program),
 * this Program is protected by copyright laws,international treaties and  other pending or existing intellectual property
 * rights in India,the United States, and other countries.Except as expressly permitted, any unauthorized reproduction,storage,
 * transmission in any form or by any means(including without limitation electronic,mechanical, printing,photocopying,
 * recording, or otherwise), or any distribution of this program, or any portion of it,may result in severe civil and
 * criminal penalties, and will be prosecuted to the maximum extent possible under the law.
 */
package com.infosys.icets.iamp.usm.domain;
import java.io.Serializable;
import lombok.EqualsAndHashCode;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Size;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

// TODO: Auto-generated Javadoc
/**
 * A UsmModule.
 */
@Entity
@Table(name = "usm_module")

/**
 * Gets the name.
 *
 * @return the name
 */
@Getter

/**
 * Sets the name.
 *
 * @param name the new name
 */
@Setter

/**
 * To string.
 *
 * @return the java.lang. string
 */
@ToString
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class UsmModule implements Serializable {

    /** The Constant serialVersionUID. */
    private static final long serialVersionUID = 1L;

    /** The id. */
    @Id
	@EqualsAndHashCode.Include
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    /** The name. */
    @Size(max = 51)
    private String name;


    // @Size(max = 256)
    // private String display_name;
    
    // @Size(max = 901)
    // private String descriptions;
    
    // @Size(max = 51)
    // private String users_count;
    
    // @Size(max = 51)
    // private String module_type;
    
    // @Size(max = 256)
    // private String url;

    
}

File: essedum-ui/shell/projects/iamp-usm/src/lib/models/usm-module.ts

export class UsmModule {
    id: number;
    name: string;
    display_name: string;
    descriptions: string;
    module_type: string;
    url: any;
    users_count: number;
    constructor(json?: any) {
     if (json != null) {
      this.id = json.id;
      this.name = json.name;
      this.display_name = json.display_name;
      this.descriptions = json.descriptions;
      this.module_type = json.module_type;
      this.url = json.url;
      this.users_count = json.users_count;
     }
    }
   
    // Utils
   
    static toArray(jsons: any[]): UsmModule[] {
     let usm_modules: UsmModule[] = [];
     if (jsons != null) {
      for (let json of jsons) {
       usm_modules.push(new UsmModule(json));
      }
     }
     return usm_modules;
    }
   }
   